### DISCUSSION TITLE
[Bug]: Text not wrapping correctly in textbox.

### DISCUSSION THREAD

**Discussion by niketaOptimumnbrew** [fabricjs/fabric.js#9747] [Q&A]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

5.3.0

### In What environments are you experiencing the problem?

Firefox, Chrome, Safari, Opera, Microsoft Edge

### Node Version (if applicable)

14.20.0

### Link To Reproduction

https://jsfiddle.net/Niketa_patel/g2kpo6xw/1/

### Steps To Reproduce

1. Please follow below code:
2. `var _wrapLine = function(_line, lineIndex, desiredWidth, reservedSpace) {
    var lineWidth = 0,
        splitByGrapheme = this.splitByGrapheme,
        graphemeLines = [],
        line = [],
        // spaces in different languges?
        words = splitByGrapheme ? fabric.util.string.graphemeSplit(_line) : _line.split(this._wordJoiners),
        word = '',
        offset = 0,
        infix = splitByGrapheme ? '' : ' ',
        wordWidth = 0,
        infixWidth = 0,
        largestWordWidth = 0,
        lineJustStarted = true,
        additionalSpace = splitByGrapheme ? 0 : this._getWidthOfCharSpacing();
        
    reservedSpace = reservedSpace || 0;
    desiredWidth -= reservedSpace;
    for (var i = 0; i < words.length; i++) {
        // i would avoid resplitting the graphemes
        word = fabric.util.string.graphemeSplit(words[i]);
        wordWidth = this._measureWord(word, lineIndex, offset);
        offset += word.length;
        
        // Break the line if a word is wider than the set width
        if (this.breakWords && wordWidth >= desiredWidth) {
        
        	if (!lineJustStarted) {
            	line.push(infix);
                lineJustStarted = true;
            }
        	
            // Loop through each character in word
            for (var w = 0; w < word.length; w++) {
            	var letter = word[w];
                var letterWidth = this.getMeasuringContext().measureText(letter).width * this.fontSize / this.CACHE_FONT_SIZE;
                if (lineWidth + letterWidth > desiredWidth) {
                	graphemeLines.push(line);
                    line = [];
                    lineWidth = 0;
                } else {
                	line.push(letter);
                    lineWidth += letterWidth;
                }
            }
            word = [];
        } else {
        	lineWidth += infixWidth + wordWidth - additionalSpace;
        }

        if (lineWidth >= desiredWidth && !lineJustStarted) {
            graphemeLines.push(line);
            line = [];
            lineWidth = wordWidth;
            lineJustStarted = true;
        } else {
            lineWidth += additionalSpace;
        }

        if (!lineJustStarted) {
            line.push(infix);
        }
        line = line.concat(word);

        infixWidth = this._measureWord([infix], lineIndex, offset);
        offset++;
        lineJustStarted = false;
        // keep track of largest word
        if (wordWidth > largestWordWidth && !this.breakWords) {
            largestWordWidth = wordWidth;
        }
    }

    i && graphemeLines.push(line);

    if (largestWordWidth + reservedSpace > this.dynamicMinWidth) {
        this.dynamicMinWidth = largestWordWidth - additionalSpace + reservedSpace;
    }

    return graphemeLines;
};

fabric.util.object.extend(fabric.Textbox.prototype, {
   _wrapLine: _wrapLine,
});

var canvas = new fabric.Canvas("c");
var str = 'Please type a long word and watch me break!';
var textbox = new fabric.Textbox(str, {
    width: 200,
    breakWords: true,
    fontSize:75
});
canvas.add(textbox);

3.

### Expected Behavior

[
![image](https://github.com/fabricjs/fabric.js/assets/126561906/be2700b5-49ab-458f-9864-6fabfc1784a2)
](url)

- Text string: "ADD HEADING TEXT"
- When the font size is 70 or above, ensure that all the words need to be visible without cutting any of them.

### Actual Behavior

[
![image](https://github.com/fabricjs/fabric.js/assets/126561906/17b1df69-9b7a-455e-b0c7-c6a83bc99e7c)
](url)

- Fontsize: 75 ( you can try with 75 or above)
- Here, two words are being cut: "H" and "N" in the textbox.

### Error Message & Stack Trace

```bash
-
```

**Comment by asturur** [comment]
@niketaOptimumnbrew are you using your own text wrapping code?

**Comment by niketaOptimumnbrew** [comment]
> @niketaOptimumnbrew are you using your own text wrapping code?

@asturur Yes. Is there any other way to wrap text in fabric.js textbox?

**Comment by asturur** [comment]
If you are using your own code and asking for help to make it work is not an issue, i ll move it to a discussion in Q & A

**Reply by niketaOptimumnbrew** [reply]
"I didn't write my own code; I followed the code provided in your answer on Stack Overflow."
Please check below link: 
https://stackoverflow.com/questions/37858237/textbox-of-fabric-js-does-not-wrap-a-long-word
